main.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+)
+
+type Document struct {
+	ID   string                 `json:"id""`
+	Data map[string]interface{} `json:"data""`
+}
+
+type DocumentStore struct {
+	documents map[string]*Document
+}
+
+func NewDocumentStore() *DocumentStore {
+	return &DocumentStore{
+		documents: make(map[string]*Document),
+	}
+}
+
+func (ds *DocumentStore) Create(document Document) error {
+	if _, ok := ds.documents[document.ID]; ok {
+		return fmt.Errorf("ID %s already exists", document.ID)
+	}
+
+	ds.documents[document.ID] = &document
+	return nil
+}
+
+func (ds *DocumentStore) ReadDocument(id string) (*Document, error) {
+	document, ok := ds.documents[id]
+
+	if !ok {
+		return nil, fmt.Errorf("document with ID '%s' does not exist", id)
+	}
+	return document, nil
+}
+
+func (ds *DocumentStore) Update(document Document) error {
+	if _, ok := ds.documents[document.ID]; !ok {
+		return fmt.Errorf("document with ID '%s' does not exist", document.ID)
+	}
+
+	ds.documents[document.ID] = &document
+	return nil
+}
+
+// SaveToFile -> takes fileName -> Save documentStore to file with given filename
+// LoadFromFile -> filename -> Load document to file with given filename
+
+func (ds *DocumentStore) SaveToFile(fileName string) error {
```

Comment:
@ujjawal-khare-27 I think you should use the functions which I am writing to write & read from files.

Reply:
@ujjawal-khare

Reply:
Checking, will do today